File: Source/typescript/backend/features/FeatureDefinition.ts
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { Field, ObjectType } from 'type-graphql';
import { IFeatureDefinition, IFeatureToggleDefinition } from '@dolittle/vanir-features';
import { FeatureToggleDefinition } from './FeatureToggleDefinition';

@ObjectType()
export class FeatureDefinition implements IFeatureDefinition {
    @Field()
    name!: string;

    @Field()
    description!: string;

    @Field(() => [FeatureToggleDefinition])
    toggles!: IFeatureToggleDefinition[];
}

File: Source/DotNET/Backend/Features/FeatureDefinition.cs
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Dolittle.Vanir.Backend.Features
{
    public class FeatureDefinition
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public FeatureToggleDefinition[] Toggles { get; set; }
    }
}
